### ISSUE TITLE
Add errorMessage property to SpinButton

### ISSUE THREAD

**Issue by mmcelroy90** [OfficeDev/office-ui-fabric-react#10241]
#### Describe the feature that you would like added

errorMessage (see errorMessage on TextField) property to be added to the SpinButton

#### What component or utility would this be added to

SpinButton

#### Have you discussed this feature with our team, and if so, who

N/A

#### Additional context/screenshots

N/A

**Comment by kenotron** [comment]
Adding a errorMessage doesn't make sense in a Spin Button control. If you wanted to display an error message, I suggest using a separate label of some sort to display that. SpinButton is just a input field with a couple of buttons going up and down.

**Comment by mmcelroy90** [comment]
We have had to write a wrapper around TextField to act as a NumericField, and this has caused us a lot of headaches over the past couple of months. We have a min and max value for all of the fields we use, but we want to display an error such as 'Value cannot be greater than $2620' while leaving their input alone and just disabling the form if there are any validation errors.

Most of the things we'd want from a numeric field are part of the SpinButton (min/max, precision etc.), but we still want to be able to handle values that are over the min/max with an error message, not just resetting the spinner to a certain value, and currently I don't see a way for the SpinButton to do that.

I guess I disagree with the notion that it doesn't make sense for a SpinButton/Numeric field to have an error message.

**Comment by kenotron** [comment]
I don't agree that we would add one off error message for every single input type. It would bloat our code a lot. A better approach is have a Form / decorator combo (much like what is happening here:
https://ant.design/components/form/).

I'm putting this whole thing as a feature request

**Comment by micahgodbolt** [comment]
This is a duplicate of a long standing need to unify our error messages #3620 which is also captured in this meta issue #7636

We'd like ALL form controls to have consistent labels, error messages etc, and would rather not do 1 off fixes.
